dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -320,7 +322,7 @@ void WebdatasetLoader::PrepareMetadataImpl() {
         for (auto& output : ext_map[component.ext]) {
           if (!was_output_set[output]) {
             DALI_ENFORCE(
-                component.size % dtypes_[output].size() == 0,
+                component.size % dtype_sizes_[output] == 0,
```

Comment:
why not just call TypeTable here? creating dtype_sizes_ produces unnecessary code, and I believe that it's not that much of a performance penalty since for example reallocating vectors or reading component data is much more expensive

Reply:
Creating dtype_sizes_ once per dataset is much cheaper than querying the type table by dynamic id over and over again for each sample.